pyproject.toml

```diff
@@ -24,11 +24,13 @@ dynamic = ["version"]
 [project.optional-dependencies]
 docs = [
     "canonical-sphinx-extensions",
+    # pre-release `furo` that supports Sphinx 8
+    #"git+https://github.com/pradyunsg/furo.git@2d3b6baa3371451d997d879a0aa61b3592ade638",
     "furo",
     "linkify-it-py",
     "myst-parser",
     "pyspelling",
-    "sphinx==6.2.1",
+    "sphinx ~= 8.0.0",
```

Comment:
Should we lock in an exact version to avoid nuisance updates/changes, e.g., `8.0.2` (latest at this time)?

Reply:
I don't think we need to.

docs/requirements.txt acts as a lock file.

A developer has to explicitly run `tox -e docs-deps` to regenerate that file and then open a PR for newer version to kick in. I don't expect nuisance here.

Meanwhile, dependabot is not aware of the causative connection between the two files. Restricting the version here won't help.

Reply:
`~= 8.0.0` will allow docs-deps to upgrade to 8.1 and 8.2 (which should be compatible), but not 9.0, right? In that case, I think what you're saying makes sense.

Reply:
> `~= 8.0.0` will allow docs-deps to upgrade to 8.1 and 8.2 (which should be compatible), but not 9.0, right? 

I think the [compatibility clause](https://packaging.python.org/en/latest/specifications/version-specifiers/#compatible-release) is tighter than that because of the extra `.0`. If I understand correctly, `~=8.0.0` is the same as `>=8.0.0 and ==8.0.*`. For what you describe it should be `~8.0`.

It's an odd specification since if these are semver then 8.1.0 should be "compatible with" 8.0.0. But my understanding is that it's saying "compatible with 8.0" rather than "compatible with 8", even though it ought to have the same effect.

Reply:
TBH I'm fine with either `~= 8.0` or `~= 8.0.0`.

I suspect it doesn't matter which, given 6.x was used before.